### DISCUSSION TITLE
Need to show 404 page

### DISCUSSION THREAD

**Discussion by saleemhaq** [OrchardCMS/OrchardCore#12613] [Q&A]
The correct url is: https://localhost:44300/article/register

![image](https://user-images.githubusercontent.com/5202291/194904756-43a316a2-2856-4ed8-8a44-9dc80e417a72.png)

but when someone edit or mistakenly type the wrong url the error has come header is twice show and error is show: An error occurred while executing this request.

![image](https://user-images.githubusercontent.com/5202291/194904090-5edf87fd-e567-4677-b55d-122ace7f1254.png)

Need to show the custom 404 page, please advise @Skrypt @ns8482e

**Comment by lampersky** [comment]
you can create custom 404 page by adding NotFound.liquid to the **Views/shared** directory

[example1](https://github.com/OrchardCMS/OrchardCore/blob/main/src/OrchardCore.Themes/TheBlogTheme/Views/Shared/NotFound.liquid)

[example2](https://github.com/OrchardCMS/OrchardCore/blob/main/src/OrchardCore.Themes/TheAgencyTheme/Views/Shared/NotFound.liquid)

**Reply by saleemhaq** [reply]
> you can create custom 404 page by adding NotFound.liquid to the **Views/shared** directory
> 
> [example1](https://github.com/OrchardCMS/OrchardCore/blob/main/src/OrchardCore.Themes/TheBlogTheme/Views/Shared/NotFound.liquid)
> 
> [example2](https://github.com/OrchardCMS/OrchardCore/blob/main/src/OrchardCore.Themes/TheAgencyTheme/Views/Shared/NotFound.liquid)

I already has the 404 page but it was not show

![image](https://user-images.githubusercontent.com/5202291/194918808-36177dd2-04ae-458c-96d4-a415c39333fe.png)

I used the razor syntax and simple line show the form and i am using the theme in decouple cms see below:

![image](https://user-images.githubusercontent.com/5202291/194919316-2849ee4e-4096-418a-907c-7e03c59c1dad.png)

Please advise

**Comment by jtkech** [comment]
You are checking if `article` is null or not but then further you are using it as is `article.Content...` so it fails on server side, first just try to use a null check e.g. with the null operator `article?.Content...`, and without casting it to `(string`) as it may fail if the result is null.

Then maybe something like this

    var markdown = article?.As<MarkdownBodyPart>()?.Markdown;

**Reply by saleemhaq** [reply]
> You are checking if `article` is null or not but then further you are using it as is `article.Content...` so it fails on server side, first just try to use a null check e.g. with the null operator `article?.Content...`, and without casting it to `(string`) as it may fail if the result is null.
> 
> Then maybe something like this
> 
> ```
> var markdown = article?.As<MarkdownBodyPart>()?.Markdown;
> ```

Want to show the 404 page if article is null so how can i achieve that @jtkech

**Comment by ns8482e** [comment]
@saleemhaq If I were you, l’d have used OnGet using PageModel and inside OnGet if article is null returned NotFound that would have captured by my 404 page

**Reply by saleemhaq** [reply]
> @saleemhaq If I were you, l’d have used OnGet using PageModel and inside OnGet if article is null returned NotFound that would have captured by my 404 page

Can you shared with the example code for that in razor syntax i will try to applied that @ns8482e

**Reply by ns8482e** [reply]
```cs
public IActionResult OnGet(int id)
{

//query your content item from id
    var markdown = article?.As<MarkdownBodyPart>()?.Markdown;
    if (markdown == null)
    {
        return NotFound();
    }

    return Page();
}
```
